### PR TITLE
DUPLO-29372 TF:ASG: Getting 'context deadline exceeded' for resource with deprecated field 'zone'

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

... (don't forget to put your clickup ticket ID in the title) ...

## Summary of changes
Fixed timeout issue for resource duplocloud_asg_profile
This PR does the following:

- Handled zone and zones field initialization
- ...

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✔︎] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

File: duplocloud/resource_duplo_asg_profile.go [hunk 1]
```diff
@@ -409,12 +409,18 @@ func asgProfileToState(d *schema.ResourceData, duplo *duplosdk.DuploAsgProfile)
 
 	// If a network interface was customized, certain fields are not returned by the backend.
 	if v, ok := d.GetOk("network_interface"); !ok || v == nil || len(v.([]interface{})) == 0 {
-		if len(duplo.Zones) > 0 {
+		_, zok := d.GetOk("zone")
+
+		if len(duplo.Zones) > 0 && !zok {
```
Comment (pravin-duplocloud) [inline duplocloud/resource_duplo_asg_profile.go:414]: do we need to check duplicates

**Review by pravin-duplocloud — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_asg_profile.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_asg_profile.go:414]: Its not duplicate check. 
Since zone is depreciated, and backend only accept zones, this logic is done for backward compatibility and to avoid diff on zones.

Here logic is to check if user passed value through zone field or zones field. Both cannot be passed at once implemented conflictsWith attribute for the fields

**Review by Matheusbafutto — APPROVED** [review]

File: duplocloud/resource_duplo_asg_profile.go (hunk 1 above)
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_asg_profile.go:414]: i mean duplicates in before appending  to
for _, v := range duplo.Zones {
	 i = append(i, v)
}
mostly BE will neglect duplicate zones

Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_asg_profile.go:414]: can you test corner case above before merge

**Review by pravin-duplocloud — APPROVED** [review]

File: duplocloud/resource_duplo_asg_profile.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_asg_profile.go:414]: ok

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_asg_profile.go:414]: There is BE validation in place for it attaching screenshot

Screenshot 2025-01-17 at 10 27 55

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_asg_profile.go:414]: tested cases
set only zone ==> resource gets created
set only zones ==> resource gets created
set both zone and zones ==> conflicts
without zone and zones ==> resource gets created
duplicate values in zones ==> BE has validation check for it

**Review by pravin-duplocloud — APPROVED** [review]
